### PR TITLE
Add a traceSampled field to the google-cloud-logging LogEntry class.

### PR DESCRIPTION
**Pull request by sebright2**
The new field was added to the Stackdriver LogEntry protocol buffer message in
https://github.com/googleapis/googleapis/commit/7b34e5a3ed10775fe14d15ffa842870dc3b5400e#diff-a51916a680df6453b6d86586cd7d63f9R149.

______________________________________________________________________________

This change would allow [opencensus-contrib-log-correlation-stackdriver](https://github.com/census-instrumentation/opencensus-java/tree/master/contrib/log_correlation/stackdriver) to set the trace_sampled field in its LoggingEnhancer.

/cc @g-easy

### CODE REVIEW

File: google-cloud-clients/google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java [hunk 1]
```diff
@@ -505,6 +529,7 @@ private static Long millisFromTimestamp(Timestamp timestamp) {
     if (spanId != null) {
       builder.setSpanId(spanId);
     }
+    builder.setTraceSampled(traceSampled);
```
Comment (pongad) [inline google-cloud-clients/google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java]: Do we need to null check this?

Comment (sebright) [reply google-cloud-clients/google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java]: The protocol buffer LogEntry.Builder's traceSampled field is a boolean ([link](https://github.com/googleapis/google-cloud-java/blob/f69e7fafe748bc8ce49c2a8d9375b3b3a4644d5d/google-api-grpc/proto-google-cloud-logging-v2/src/main/java/com/google/logging/v2/LogEntry.java#L4141)), so I thought that it wouldn't be necessary to distinguish between false and not set.  The traceSampled field in this class is also a boolean.

Comment (pongad) [reply google-cloud-clients/google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java]: Gah you're right. Sorry I have been working with Apiary for too long!

**Comment by g-easy** [comment]
Consider adding wording to the description along the lines of "so that the backend can tell the difference between sampled and non-sampled traces, to e.g. link from the log view to the trace view for them"

**Comment by sebright** [comment]
I tried to keep the description short, since I think the protocol buffer is the source of truth.  
The descriptions for `getTrace` and `getSpanId` are also relatively short.  Would it be better to copy the field description from the LogEntry proto?

**Comment by g-easy** [comment]
Up to you. I'm fine with keeping it short.

**Comment by sebright** [comment]
Thanks!

**Pull request merged**